Onboarding/Connector/Scripts/Enable-ImpactReporting.ps1

```diff
@@ -0,0 +1,287 @@
+          "connectorType" = "AzureMonitor"
+      }
+  } | ConvertTo-Json
+
+  $QueryParam="?api-version=2024-05-01-preview"
+  $Path = "subscriptions/$SubscriptionId/providers/Microsoft.Impact/connectors/$ConnectorName$QueryParam"
+
+  $AttemptCount = 1
+  $SleepDuration = 10
+  $ProvisioningState=""
+  do
+  {
+      $ResponseBody = (Invoke-AzRestMethod -Method Put -Path $Path -Payload $request_body).Content | ConvertFrom-Json
+      $ProvisioningState = $ResponseBody.properties.provisioningState 
+      if ($ProvisioningState -ne "Succeeded")
+      {
+        $SleepDuration = $SleepDuration * 2
+        Log "Attempt #${AttemptCount}: Connector deployment is in progress. Waiting for $SleepDuration seconds before retrying."
+        Start-Sleep -Seconds $SleepDuration
+        $AttemptCount = $AttemptCount + 1
+      }
+      else
+      {
+        Log "Attempt #${AttemptCount}: Deployment of connector: $ConnectorName is successful."
+      }
+      
+  } while ($ProvisioningState -ne "Succeeded")
+}
+
+function Test-Input {
+  param (
+      [string]$SubscriptionId,
+      [string]$FilePath
+  )
+
+  if ((-not $SubscriptionId) -and (-not $FilePath)) {
+      Log "Please provided required input: SubscriptionId or FilePath with list of subscription IDs while invoking the script"
+      exit 1
+  }
+  if ($SubscriptionId -and $FilePath) {
+      Log "Please provided either SubscriptionId or FilePath with list of subscription IDs while invoking the script. Do not provide both."
+      exit 1
+  }
+
+  if ($SubscriptionId) {
+      Log "Using Subscription ID = $SubscriptionId"
+  }
+
+  if ($FilePath) {
+      if (-not (Test-Path -Path $FilePath)) {
+          Log "Failed to find file: $FilePath. Please provide a valid file path that you have access to"
+          exit 1
+      }
+      Log "Subscription IDs will be read from the file: $FilePath"
+  }
+
+  Log "==== All required inputs are present ===="
+}
+
+function Enable-ImpactReporting {
+  param (
+      [array]$SubscriptionIds
+  )
+  # Loop through the array
+  foreach ($item in $SubscriptionIds) {
+      LoginAzWithPrompt $item
+      Register-Feature "Microsoft.Impact" "AzureImpactReportingConnector"
+      Register-Feature "Microsoft.Impact" "AllowImpactReporting"
+      Add-PermissionsForAlertReading $item
+      Deploy-Connector $item
+      Log "==== Impact reporting is now enabled on your subscription: $item!! ===="
+  }
+}
+
+if ($help) {
+  PrintHelp
+  exit
+}  
+
+Test-Input -SubscriptionId $SubscriptionId -FilePath $FilePath
+
+# Check if the file is given and exists
+if ($FilePath -and (Test-Path -Path $FilePath)) {
+  # Read the file into an array
+  $SubscriptionIds = Get-Content -Path $FilePath
+}
+else {
+  # Create an array with a single element
+  $SubscriptionIds = @($SubscriptionId)
+}
+
+Log "==== Enabling impact reporting ===="
+Enable-ImpactReporting -SubscriptionIds $SubscriptionIds
+Log "==== Impact reporting is successfully enabled on your subscription(s) ===="
```

Comment:
Consider rephrasing to "Impact Reporting Connector is successfully created in your subscription."